### DISCUSSION TITLE
Github Pages can't find Tailwind Css

### DISCUSSION THREAD

**Discussion by kingpixel** [tailwindlabs/tailwindcss#2722] [Help]
I have a static jekyll site hosted on github pages. i added Tailwind as a dependency and it builds fine locally, but when push my source to master, the build fails with this error: 

Your SCSS file `assets/css/style.scss` has an error on line 8: File to import not found or unreadable: tailwindcss/base. Load path: /hoosegow/.bundle/ruby/2.7.0/gems/jekyll-theme-slate-0.1.1/_sass.

I googled the heck out of this, and I don't know how to fix this. Any ideas? 
thanks in advance!

**Comment by simonswiss** [comment]
Hey!

Any chance you could link to the repo in question - or (even better) recreate a minimal reproduction of the problem?

This definitely sounds like a Jekyll config issue, in the likes of this one: https://stackoverflow.com/a/51550090

**Reply by kingpixel** [reply]
I'm assuming the tailwind assets are in the node_modules directory, but I don't know how to get github pages/Jekyll to find them. As I said, this works fine locally. Here is the repo https://github.com/kingpixel/kingpixel.github.io

**Reply by simonswiss** [reply]
Hey, thanks for that.

I am able to reproduce the error in GitHub pages, but when I build locally with `bundle exec jekyll build` and deploy the `_site` directory anywhere (local server, netlify etc) everything works perfectly fine.

![image](https://user-images.githubusercontent.com/485747/97818962-b8ee6e00-1cf9-11eb-9b48-ff1987db6ea8.png)

This is most definitely an issue with how GitHub Pages and Jekyll work and handle the site build - have you gone through all the scenarios from [that page](https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/troubleshooting-jekyll-build-errors-for-github-pages-sites#invalid-sass-or-scss)?

**Reply by simonswiss** [reply]
Update: this might very well solve your problem.

https://stackoverflow.com/a/43487798

> Github pages uses a version of Jekyll that ignores node_modules folder by default.

would explain while the `@tailwind/base` cannot be imported in the scss file to build the site 😅

> Finally, to make it easier to vendor third-party dependencies via package managers like Bundler or NPM (or Yarn), Jekyll now ignores the vendor and node_modules directories by default, speeding up build times and avoiding potential errors. If you need those directories included in your site, set exclude: [] in your site’s configuration file.

**Reply by matt-scott** [reply]
I am having the exact same issue. I am able to build and view my site locally but get the same error about the tailwindcss files when I upload to github. I added the node_modules folder to my repository, but that doesn't seem to have done anything. I also added .css to the ends of the three imports (base, components, and utilities), per another post (which I've since lost from my browser tabs) to no avail.

Did you ever figure this out? Any other ideas on this?

**Reply by matiszz** [reply]
Hey there! Any updates on this? I'm having exactly the same problem: Jekyll blog-site with Tailwind CSS. Works fine locally, but it gives me this error when pushing it to `main`, where I have my gh-pages:
![image](https://user-images.githubusercontent.com/9436778/115624265-ac3af200-a2fa-11eb-800e-ce192a7ef087.png)
Things I've tried:
- Adding my `node_modules` directory to git.
- Adding `exclude: []` to my `_config.yml`
- Adding the sass location with `jekyll-sass-converter`:
   ```
    sass:
      sass_dir: _sass/
      load_paths:
        - ../node_modules/
     ```

Any help please? 🙏🏽 
Link to my repo: https://github.com/matiszz/matias-szarfer

**Reply by matiszz** [reply]
Well, I finally reverse-engineered @kingpixel's site, and saw [this](https://github.com/kingpixel/kingpixel.github.io/commit/45ba10a690a87d80cfb8aaa26c71414fc0657885) commit. I did the same, changing the node install for a CDN and worked fine.

Add this to the `head` your `default.html`
```
    <link href="https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css" rel="stylesheet">
```
